diff --git a/src/wrapasvst3.cpp b/src/wrapasvst3.cpp
index 12a6493..4635022 100644
--- a/src/wrapasvst3.cpp
+++ b/src/wrapasvst3.cpp
@@ -319,0 +320,5 @@ tresult PLUGIN_API ClapAsVst3::setupProcessing(Vst::ProcessSetup &newSetup)
+  if (!(newSetup.sampleRate == newSetup.sampleRate) || newSetup.sampleRate < 8000.0 ||
+      newSetup.sampleRate > 192000.0 || newSetup.maxSamplesPerBlock == 0)
+  {
+    return kResultFalse;
+  }
@@ -939 +944 @@ void ClapAsVst3::setupWrapperSpecifics(const clap_plugin_t *plugin)
-  _useIMidiMapping = checkMIDIDialectSupport();
+  _useIMidiMapping = (_plugin->_ext._params != nullptr) && checkMIDIDialectSupport();
