diff --git a/src/wrapasvst3.h b/src/wrapasvst3.h
index 5b15e43..861adba 100644
--- a/src/wrapasvst3.h
+++ b/src/wrapasvst3.h
@@ -249 +249 @@ class ClapAsVst3 : public Steinberg::Vst::SingleComponentEffect,
-  if (::Steinberg::FUnknownPrivate::iidEqual(iid, IMidiMapping::iid))
+  if (::Steinberg::FUnknownPrivate::iidEqual(_iid, IMidiMapping::iid))
@@ -257 +257 @@ class ClapAsVst3 : public Steinberg::Vst::SingleComponentEffect,
-  if (::Steinberg::FUnknownPrivate::iidEqual(iid, IContextMenuTarget::iid))
+  if (::Steinberg::FUnknownPrivate::iidEqual(_iid, IContextMenuTarget::iid))
@@ -264 +264 @@ class ClapAsVst3 : public Steinberg::Vst::SingleComponentEffect,
-  if (::Steinberg::FUnknownPrivate::iidEqual(iid, IPlugInEntryPoint::iid))
+  if (::Steinberg::FUnknownPrivate::iidEqual(_iid, IPlugInEntryPoint::iid))
@@ -271 +271 @@ class ClapAsVst3 : public Steinberg::Vst::SingleComponentEffect,
-  if (::Steinberg::FUnknownPrivate::iidEqual(iid, IPlugInEntryPoint2::iid))
+  if (::Steinberg::FUnknownPrivate::iidEqual(_iid, IPlugInEntryPoint2::iid))
@@ -278 +278 @@ class ClapAsVst3 : public Steinberg::Vst::SingleComponentEffect,
-  if (::Steinberg::FUnknownPrivate::iidEqual(iid, Presonus::IGainReductionInfo::iid))
+  if (::Steinberg::FUnknownPrivate::iidEqual(_iid, Presonus::IGainReductionInfo::iid))
@@ -285 +285 @@ class ClapAsVst3 : public Steinberg::Vst::SingleComponentEffect,
-  if (::Steinberg::FUnknownPrivate::iidEqual(iid, Steinberg::Vst::ChannelContext::IInfoListener::iid))
+  if (::Steinberg::FUnknownPrivate::iidEqual(_iid, Steinberg::Vst::ChannelContext::IInfoListener::iid))
@@ -294 +294 @@ class ClapAsVst3 : public Steinberg::Vst::SingleComponentEffect,
-  //if (::Steinberg::FUnknownPrivate::iidEqual(iid, IExampleSomething::iid))
+  //if (::Steinberg::FUnknownPrivate::iidEqual(_iid, IExampleSomething::iid))
